tests/infra/node.py

```diff
@@ -305,47 +310,83 @@ def _start(
-            )
-            with open(node_address_file, "r", encoding="utf-8") as f:
-                node_host, node_port = f.read().splitlines()
-                node_port = int(node_port)
-                if self.remote_shim != infra.remote_shim.DockerShim:
-                    assert (
-                        node_host == self.node_host
-                    ), f"Unexpected change in node address from {self.node_host} to {node_host}"
-                if self.node_port != 0:
-                    assert (
-                        node_port == self.node_port
-                    ), f"Unexpected change in node port from {self.node_port} to {node_port}"
-                    self.node_port = node_port
-                self.node_port = node_port
-
-        if self.remote.rpc_addresses_file is not None:
-            rpc_address_file = os.path.join(
-                self.common_dir, self.remote.rpc_addresses_file
-            )
-            with open(rpc_address_file, "r", encoding="utf-8") as f:
-                lines = f.read().splitlines()
-                it = [iter(lines)] * 2
-            for (rpc_host, rpc_port), rpc_interface in zip(
-                zip(*it), self.host.rpc_interfaces
-            ):
-                rpc_port = int(rpc_port)
-                if self.remote_shim != infra.remote_shim.DockerShim:
-                    assert (
-                        rpc_host == rpc_interface.rpc_host
-                    ), f"Unexpected change in RPC address from {rpc_interface.rpc_host} to {rpc_host}"
-                if rpc_interface.rpc_port != 0:
-                    assert (
-                        rpc_port == rpc_interface.rpc_port
-                    ), f"Unexpected change in RPC port from {rpc_interface.rpc_port} to {rpc_port}"
-                rpc_interface.rpc_port = int(rpc_port)
-                # In the infra, public RPC port is always the same as local RPC port
-                rpc_interface.public_rpc_port = rpc_interface.rpc_port
+        if self.major_version is None or self.major_version > 1:
+            if self.remote.node_address_file is not None:
+                node_address_file = os.path.join(
+                    self.common_dir, self.remote.node_address_file
+                )
+                self._resolve_address(
+                    node_address_file,
+                    {infra.interfaces.NODE_TO_NODE_INTERFACE_NAME: self.n2n_interface},
+                )
+
+            if self.remote.rpc_addresses_file is not None:
+                rpc_address_file = os.path.join(
+                    self.common_dir, self.remote.rpc_addresses_file
+                )
+                self._resolve_address(rpc_address_file, self.host.rpc_interfaces)
+                #  In the infra, public RPC port is always the same as local RPC port
+                for _, interface in self.host.rpc_interfaces.items():
+                    interface.public_port = interface.port
+        else:
+            # Legacy 1.x nodes
```

Comment:
Because of the LTS compatibility test, we need to keep this code around since we spin up 1.x nodes that output RPC addresses and node address files in a different format. I'm tempted to say we should stop doing this once 2.0 is out and instead generate and store a 1.x ledger somewhere that we can easily download from the LTS compatibility test so we don't have to spin up old nodes anymore.

Reply:
I've added a note in https://github.com/microsoft/CCF/issues/2981#issue-996935226.